mlir/lib/Conversion/TosaToRock/TosaToRock.cpp

```diff
@@ -811,16 +810,31 @@ struct AttentionRewritePattern : public OpRewritePattern<tosa::MatMulOp> {
     if (failed(softmaxInput)) {
       return failure();
     }
-    if (tosa::MulOp scale =
-            getDefiningNonReshapeOp<tosa::MulOp>(softmaxInput.value())) {
+
+    Value group = nullptr;
```

Comment:
nit naming : what is meant by group here ?

Reply:
Yes, the naming is ambiguous.

Here is what used to be.
```
Softmax(Q * K .* Scale)
        ^^^^^^^^^^^^^
  used to be called softmax input
```

Now, we are dealing with this scenario. Basically, I needed a variable which could hold  `Q * K .* Scale` sub-expression. I just called this a group. 

```
Softmax(Q * K .* Scale + Bias)
        ^^^^^^^^^^^^^
            group
        ^^^^^^^^^^^^^^^^^^^^^^
             softmax input
```

Sorry, no better name came into my mind. What do you think?

Reply:
why not scaledMatMul ?

Reply:
sure